sdk/core/azure-core/inc/azure/core/context.hpp

```diff
@@ -391,12 +391,17 @@ namespace Azure { namespace Core {
           = ContextSharedState::ToMsecSinceEpoch(time_point::min());
     }
 
+    /**
+     * @brief Checks if the context was canceled.
+     */
+    bool IsCanceled() const { return CancelWhen() < std::chrono::system_clock::now(); }
```

Comment:
Tests need to be added for this new method.

Reply:
I expected existing unit tests for `ThrowIfCanceled` to also test this method indirectly, but looks like we don't have tests for that method either. :)

Reply:
Oh... no tests or even any callers for `WithDeadline()` :(

Reply:
https://github.com/Azure/azure-sdk-for-cpp/issues/1146